gapis/api/vulkan/errors.api

```diff
@@ -57,6 +57,7 @@ extern void vkErrInvalidImageSubresource(VkImage img, string subresourceType, u3
 extern void vkErrImageMemoryNotBound(VkImage img)
 extern void vkErrSemaphoreNotSubmitted(VkSemaphore semaphore)
 extern void vkErrInvalidDescriptorCopy(VkDescriptorSet srcSet, u32 srcBinding, VkDescriptorSet dstSet, u32 dstBinding)
+extern void vkErrInvalidStructMember(string structName, string structMember)
```

Comment:
I don't understand this error message. From where this error is used, it sound like the value is bad, but we are only showing the name of the field? I think it makes sense to add a specific error message for this case and show both the expected value and the passed in value. I think we've been trying to use more, but specific errors in favor of less, but more general errors.

Reply:
I don't like it either. It's awkward because the string formatting is separated from the error location so each unique error message has to be plumbed through a handful of files. I was trying to make something that could be reused but I agree it is almost too vague to be useful. What about something like this:

```suggestion
extern void vkErrInvalidValueU32(string name, u32 expected, u32 actual)
```

I'm second-guessing if we really want this error at all. It isn't fatal to the capture or replay operations and putting API validation in the .api files feels out of place.

Reply:
Yeah, I think you're right, drop the error all-together. We've only really been minimal validation, such as null checks, so let's not start with this one. In general, it's a requirement that the app has no validation errors for tracing.